pydocstringformatter/formatting/base.py

```diff
@@ -47,3 +49,42 @@ def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
             tokeninfo.end,
             tokeninfo.line,
         )
+
+
+class StringAndQuotesFormatter(Formatter):
+    """Base class for string formatter that needs access to the quotes."""
+
+    quotes_regex = re.compile(r"""['"]{1,3}""")
+    """Pattern to match against opening quotes."""
+
+    @abc.abstractmethod
+    def _treat_string(
+        self,
+        tokeninfo: tokenize.TokenInfo,
+        indent_length: int,
+        quotes: str,
+        quotes_length: Literal[1, 3],
+    ) -> str:
+        """Return a modified string."""
+
+    def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
+        # Get the quotes used for this docstring
+        match = re.match(self.quotes_regex, tokeninfo.string)
+        assert match
+        quotes = match.group()
+
+        quotes_length = len(quotes)
+        assert quotes_length in {1, 3}
+
+        return tokenize.TokenInfo(
+            tokeninfo.type,
+            self._treat_string(
+                tokeninfo,
+                tokeninfo.start[1],
+                quotes,
+                quotes_length,  # type: ignore[arg-type]
```

Comment:
Both `mypy` and `pyright` don't acknowledge that `quotes_length` is `1` or `3`. `pyright` does if you use `(1,3)`, but then `pylint` complains about set comparison.